app/public/static/js/components/ordered-puzzle.js

```diff
@@ -0,0 +1,196 @@
+AFRAME.registerComponent("ordered-puzzle", {
+  schema: {
+    jsonData: {
+      parse: JSON.parse,
+      stringify: JSON.stringify,
+    },
+    isSolved: { type: "map" },
+  },
+
+  multiple: true,
+
+  init: function () {
+    const data = this.data.jsonData;
+    const el = this.el;
+    this.id = data.id;
+    this.useTargets = data.useTargets;
+    const is_last_object =
+      "is_last_object" in data ? data.is_last_object : false;
+
+    const numPuzzlePieces = data.images.length;
+    const blackboard = document.querySelector("#blackboard");
+    this.puzzlePieceCache = [];
+
+    if (data.useTargets) {
+      // Create text label which indicates to users that they have successfully completed the puzzle
+      const solvedPuzzleText = {
+        width: "5",
+        height: "1",
+        depth: "1",
+        color: "green",
+        x: "0",
+        y: "-5.75",
+        z: "0",
+        scaleX: "4",
+        scaleY: "4",
+        scaleZ: "2",
+        text: "Solved",
+      };
```

Comment:
Correct me if I'm wrong, but these attributes are going to be the same for all solved puzzle texts right? (same position, color, scale, etc). Should we try to extract this to a generic place?

Reply:
Unfortunately, this won’t be the same for all of them, the scaling, xyz and height/width are different for this vs keypad for example.

Reply:
Ah rip, okay